src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -444,6 +444,23 @@ private void ScanInterpolatedStringLiteralContents(ArrayBuilder<Interpolation>?
 
             private void ScanFormatSpecifier()
             {
+                /*
+                ## Grammar from spec:
+                
+                interpolation_format
+                    : ':' interpolation_format_character+
+                ; 
+                interpolation_format_character
+                    : '<Any character except \" (U+0022), : (U+003A), { (U+007B) and } (U+007D)>'
+                ;
+
+                ------------------
+                Even though the spec says that ", :, { and } are prohibited in the format specifier
+                we allowed them as long as they were escaped. This caused the that the format specifier
+                in `$"{{{12:X}}}"` was parsed in a way that it had the value "X}" which
+                is not spec conformant nor what users expect.
+                 */
```

Comment:
i don't generally think this comment is that helpful.  But i also don't think it hurts too much to have.